### PR TITLE
Link fixes for docs build

### PR DESCRIPTION
**Pull request by hflynn**
Updated this link after some googling but this new one isn't working at the moment either so I've also added it to the ignore list for now - hopefully the not working bit is temporary and I'll be able to remove it from the ignore list next week. Otherwise, I'll look at using a completely different link but this will hopefully at least get the build green for today.

### CODE REVIEW

File: docs/sphinx/conf.py.in [hunk 1]
```diff
@@ -371,5 +371,9 @@ linkcheck_ignore = ['https://imspector.mpibpc.mpg.de',
     'https://support.apple.com/downloads/quicktime',
     'https://support.apple.com/quicktime',
     'https://nifti.nimh.nih.gov/nifti-1/',
-    'https://www.leica-microsystems.com'
+    'https://www.leica-microsystems.com',
```
Comment (sbesson) [inline docs/sphinx/conf.py.in]: Should we remove this link as per https://github.com/ome/ome-model/pull/66?

Comment (hflynn) [reply docs/sphinx/conf.py.in]: Good point, I'll push another commit and remove it

**Comment by hflynn** [comment]
Rebuilding merge docs with latest commit included

**Comment by hflynn** [comment]
https://ci.openmicroscopy.org/view/Docs/job/BIOFORMATS-DEV-merge-docs/1007/ still green

**Comment by hflynn** [comment]
Wil recheck status of the links on the ignore list here and see if some can be removed

**Comment by hflynn** [comment]
Manual testing suggests everything on the ignore list should work but trying to use the linkchecker is a no go just now as our own domains are failing

**Comment by hflynn** [comment]
Ran linkchecker locally - two of the ignore list still fail the linkchecker despite working manually and I added the link currently breaking the merge build because the certificate is expired. This should get the merge build green for tomorrow morning and hopefully we'll be able to remove the libjpeg-turbo link from the ignore list later this week before merging this.

**Comment by hflynn** [comment]
libjpeg certificate seems to be working again today, will remove that from the ignore list

**Comment by hflynn** [comment]
I'll update any permanent redirects on this PR too

**Comment by hflynn** [comment]
I'll list this for review tomorrow if the build is green now - we should probably get it merged asap before we end up with conflicts as I've ended up touching so many files.

**Comment by hflynn** [comment]
https://ci.openmicroscopy.org/view/Docs/job/BIOFORMATS-DEV-merge-docs/1026/ is green now

**Comment by dgault** [comment]
All of the updated links work as expected when tested manually. We also have had a couple of successive green builds on the merge-docs now.

**Comment by sbesson** [comment]
Thanks for another round of the periodic URL cleanup @hflynn.

**Comment by hflynn** [comment]
Ah, found a few permanent redirects I missed looking at the merge job log - I'll push another commit for completeness.

**Comment by hflynn** [comment]
Or not ;)

**Comment by hflynn** [comment]
I only missed two http -> https instances, I'll clean them up next time I'm fixing links.

**Pull request merged**